gammapy/spectrum/observation.py

```diff
@@ -518,6 +518,24 @@ def copy(self):
         """A deep copy."""
         return copy.deepcopy(self)
 
+    def to_spectrum_dataset(self):
```

Comment:
Indeed, it is probably much better to have this function on `SpectrumObservation` rather than on the `Dataset` itself.